### DISCUSSION TITLE
My DX is gone because of `Refreshing page data due to server-side change`

### DISCUSSION THREAD

**Discussion by italodeandra** [vercel/next.js#32792] [Help] [answered]
Hi team,

For some unknown reason, at random, when I change the code it triggers the `Refreshing page data due to server-side change`. There's no pattern, I can't reproduce nor give examples. It just happens at random and I'm losing my mind over it.

Even changing just a letter on the code triggers it. Sometimes it doesn't sometimes it does.

So the approach I was looking for to try to solve this is to understand better what triggers the so called "server-side change".

Does anyone can explain to me a little bit of how this message `Refreshing page data due to server-side change` works? I had no luck with Google or Next.js documentation.

**Comment by webmandman** [comment]
@italodeandra  Did you ever get this resolved?

**Reply by italodeandra** [reply]
Actually yes. It was an issue with one of the early patches of `v13`. It got fixed in a later version.

But just so you know, this "problem" is a normal behavior when you have issue with what you export from React component files. Please, take a look at https://nextjs.org/docs/basic-features/fast-refresh#how-it-works

**Comment by italodeandra** [comment, marked as answer]
It was an issue with one of the early patches of v13. It got fixed in a later version.
